lib/queues.js

```diff
@@ -611,5 +610,22 @@ module.exports = class Queues {
     // Lazy load configuration.
     return this._ironium.configPromise;
   }
+
+  // Returns the maximum amount of jobs a queue should try to reserve.
+  // Every queue gets a share of reservation capacity to better deal
+  // with load spikes across queues.
+  getReserveCapacity() {
+    return this.configPromise
+      .then(config => {
+        const totalCapacity       = config.concurrency - this._activeJobs;
+        const maxCapacityPerQueue = Math.floor(totalCapacity / this._queues.size);
+        const capacity            = Math.min(totalCapacity, maxCapacityPerQueue);
```

Comment:
Going from 30 to 10 means 3x as many API calls. Doesn't seem very effective, unless you have a few queues that are constantly under load. I'd expect it to be a power law distribution, where 20% of queues do 80% of the workload.